File: 841-shortest-distance-to-a-character/shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        res = [0] * len(s)
        c_idx = []

        for i in range(len(s)):
            if s[i] == c:
                c_idx.append(i)

        for i in range(len(s)):
            if len(c_idx) >= 2:
                res[i] = min(abs(c_idx[0] - i), abs(c_idx[1] - i))
                if abs(c_idx[0] - i) > abs(c_idx[1] - i):
                    c_idx = c_idx[1:]
            else:
                res[i] = abs(c_idx[0] - i)
   
        return res

File: 841-shortest-distance-to-a-character/shortest-distance-to-a-character.java
class Solution {
    public int[] shortestToChar(String s, char c) {
        int n = s.length();
        int distance = 0;
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == c) {
                distance = i;
                break;
            }
        }

        int[] result = new int[s.length()];
        for (int i = 0; i < n; i++) {
            if (c == s.charAt(i)) {
                distance = i;
            }
            result[i] = Math.abs(distance - i);
        }
        

        for (int i = n - 1; i >= 0; i--) {
            if (c == s.charAt(i)) {
                distance = i;
            }
            result[i] = Math.min(result[i], Math.abs(distance - i));
        }

        return result;
    }
}